packages/compass-aggregations/src/modules/import-pipeline.js

```diff
@@ -211,26 +213,17 @@ export const confirmNew = () => (dispatch, getState) => {
 };
 
 /**
- * Create a pipeline from the provided text.
- *
- * @param {String} text - The text.
- *
- * @returns {Array} The pipeline for the builder.
+ * @param {import('./pipeline-builder/pipeline-parser').Stage[]} stages 
+ * @returns {import('./pipeline').StageState}
  */
-export const createPipeline = (text) => {
-  try {
-    const stages = extractStages(text);
-    return stages.map((stage) => {
-      return createStage(
-        stage.operator,
-        stage.source,
-        null
-      );
-    });
-  } catch (jsError) {
-    debug(jsError);
-    return [ createStage(null, '', jsError.message) ];
-  }
+const mapBuilderStagesToUIStages = (stages) => {
+  return stages.map(({operator, value, syntaxError}) => ({
+    ...emptyStage(),
+    stageOperator: operator,
+    stage: value,
+    isValid: syntaxError ? false : true,
+    syntaxError: syntaxError?.message
+  }));
 };
```

Comment:
There is a method in utils to create stage with defaults: `generateStageWithDefaults`

I would also consider moving this whole thing to stage utils

Reply:
oh thanks! and i can reuse that :)